### PR TITLE
yajl: initial onboarding (v2.1.0)

### PR DESCRIPTION
**Pull request by gegles**
Specify library name and version:  **yajl/2.1.0**

This PR aims at adding the [YAJL library](https://github.com/lloyd/yajl) to the Conan Center.

Overall, it was a very straightforward process and I essentially was able to follow the template project. 

The looks hooks all reported OK.

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by gegles** [comment]
FYI @bemehiser

**Comment by gegles** [comment]
@prince-chrismc @RubenRBS et. al. just FYI, this one is all green and ready to go ;-) Thx!

File: recipes/yajl/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+        # We need to move the dll from lib to bin in order for it to be found later
+        if self.settings.os == "Windows":
+            rename(self, os.path.join(self.package_folder, "lib", "yajl.dll"), os.path.join(self.package_folder, "bin", "yajl.dll"))
+        fix_apple_shared_install_name(self)
+
+    def package_info(self):
+        self.cpp_info.libs = ["yajl"]
+
+        self.cpp_info.set_property("cmake_file_name", "yajl")
```
Comment (prince-chrismc) [inline recipes/yajl/all/conanfile.py:77]: Looks like some left-over template stuff? looking at upstream, https://github.com/lloyd/yajl/blob/5e3a7856e643b4d6410ddc3f84bc2f38174f2872/src/CMakeLists.txt#L48

The are not install config files so this should not be present

all the cmake stuff should be dropped

File: recipes/yajl/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        self.cpp_info.set_property("cmake_target_name", "yajl::yajl")
+        self.cpp_info.set_property("pkg_config_name", "yajl")
```
Comment (prince-chrismc) [inline recipes/yajl/all/conanfile.py:79]:
```suggestion
        # https://github.com/lloyd/yajl/blob/5e3a7856e643b4d6410ddc3f84bc2f38174f2872/src/CMakeLists.txt#L64
        self.cpp_info.set_property("pkg_config_name", "yajl")
```

File: recipes/yajl/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        copy(self, pattern="COPYING", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
+        cmake.install()
+
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rm(self, "*.la", os.path.join(self.package_folder, "lib"))
+        rm(self, "*.pdb", os.path.join(self.package_folder, "lib"))
+        rm(self, "*.pdb", os.path.join(self.package_folder, "bin"))
```
Comment (prince-chrismc) [inline recipes/yajl/all/conanfile.py:67]: Template leftovers? I only see the share being used by the project

```suggestion
        rmdir(self, os.path.join(self.package_folder, "share"))
```

File: recipes/yajl/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
+from conan import ConanFile
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy, rmdir, rm, rename
```
Comment (prince-chrismc) [inline recipes/yajl/all/conanfile.py:2]: There are no patches

```suggestion
from conan.tools.files import get, copy, rmdir, rm, rename
```

File: recipes/yajl/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
```
Comment (prince-chrismc) [inline recipes/yajl/all/conanfile.py:31]:
```suggestion
    def config_options(self):
```

File: recipes/yajl/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
```
Comment (prince-chrismc) [inline recipes/yajl/all/conanfile.py:45]: this is the default so we can skip it

```suggestion
        get(self, **self.conan_data["sources"][self.version], strip_root=True)
```

File: recipes/yajl/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.generate()
+
+    def build(self):
+        apply_conandata_patches(self)
```
Comment (prince-chrismc) [inline recipes/yajl/all/conanfile.py:52]:
```suggestion
```

**Review by prince-chrismc — COMMENTED** [review]
Handful of comment to cleanup some of the extras :) the template are very over zealous nd you need way less is most cases

**Comment by prince-chrismc** [comment]
Please do not force push 🙏 GitHub forces us to restart the review which is not fun!

File: recipes/yajl/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    def package_info(self):
+        self.cpp_info.libs = ["yajl"]
+
+        # https://github.com/lloyd/yajl/blob/5e3a7856e643b4d6410ddc3f84bc2f38174f2872/src/CMakeLists.txt#L64
+        self.cpp_info.set_property("pkg_config_name", "yajl")
+
+        if self.options.shared:
+            self.cpp_info.libs = ["yajl"]
+        else:
+            self.cpp_info.libs = ["yajl_s"]
```
Comment (prince-chrismc) [inline recipes/yajl/all/conanfile.py:74]: libs where set twice

```suggestion
        self.cpp_info.libs = ["yajl"] if self.options.shared else ["yajl_s"]

        # https://github.com/lloyd/yajl/blob/5e3a7856e643b4d6410ddc3f84bc2f38174f2872/src/CMakeLists.txt#L64
        self.cpp_info.set_property("pkg_config_name", "yajl")

```

File: recipes/yajl/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+        # TODO: to remove in conan v2 once cmake_find_package_* generators removed
+        self.cpp_info.filenames["cmake_find_package"] = "YAJL"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "yajl"
+        self.cpp_info.names["cmake_find_package"] = "YAJL"
+        self.cpp_info.names["cmake_find_package_multi"] = "yajl"
```
Comment (prince-chrismc) [inline recipes/yajl/all/conanfile.py:80]: these are not needed

```suggestion
```

**Comment by prince-chrismc** [comment]
The bot does a squash merge so everything will be one big commit in the end so it does not really matter :)
The bot also builds on master if it needs to you do not need to be on latest normally

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**